src/rprblender/export/image.py

```diff
@@ -92,84 +90,53 @@ def sync(rpr_context, image: bpy.types.Image):
+        else:
+            # loading image by pixels
+            data = np.fromiter(pixels, dtype=np.float32,
+                               count=image.size[0] * image.size[1] * image.channels)
 
-    def export_full(self, rpr_context) -> (pyrpr.Image, None):
-        """ Export the full image pixels as RPR image"""
-        if self.is_empty():
-            return None
-
-        image_key = f"{self.name}@{self.color_space}"
-
-        if image_key in rpr_context.images:
-            return rpr_context.images[image_key]
-
-        pixels = np.ascontiguousarray(np.flipud(self.pixels))
-        rpr_image = rpr_context.create_image_data(image_key, pixels)
-        rpr_image.set_name(image_key)
-
-        if self.color_space in ('sRGB', 'BD16', 'Filmic Log'):
-            rpr_image.set_gamma(2.2)
+        self.pixels = data.reshape(image.size[1], image.size[0], image.channels)
 
-        return rpr_image
+        self.name = image.name
+        self.color_space = image.colorspace_settings.name
 
-    def export_region(self, rpr_context, x1, y1, x2, y2) -> (pyrpr.Image, None):
+    def export(self, rpr_context, render_size=None, tile=((0, 0), (1, 1))):
         """ Export pixels cropped to sub-region coordinates as RPR image """
-        if self.is_empty():
-            return None
 
-        # check sub-region boundaries, just in case something went terribly wrong
-        if x1 == x2 or y1 == y2 or x1 >= self.size[0] or x2 < 0 or y1 >= self.size[1] or y2 < 0:
-            log.warn(f"Image region ({x1}; {y1})-({x2}; {y2}) has no data", self.name)
-            return None
+        image_size = self.pixels.shape[1], self.pixels.shape[0]
+        if render_size:
+            image_ratio = image_size[0] / image_size[1]
+            render_ratio = render_size[0] / render_size[1]
 
-        image_key = f"{self.name}({x1}, {y1})-({x2}, {y2})@{self.color_space}"
+            if image_ratio > render_ratio:
+                size = image_size[1] * render_ratio, image_size[1]
+            else:
+                size = image_size[0], image_size[0] / render_ratio
 
-        if image_key in rpr_context.images:
-            return rpr_context.images[image_key]
+            x1, y1 = (image_size[0] - size[0]) / 2, (image_size[1] - size[1]) / 2
+            x2, y2 = x1 + size[0], y1 + size[1]
 
-        # get pixels region
-        pixels = self.extract_pixels_region(self.pixels, x1, y1, x2, y2)
-        pixels = np.flipud(pixels)
+        else:
+            x1, y1 = 0, 0
+            x2, y2 = image_size
 
-        rpr_image = rpr_context.create_image_data(image_key, np.ascontiguousarray(pixels))
+        x1, y1, x2, y2 = (x1 + (x2 - x1) * tile[0][0],
+                          y1 + (y2 - y1) * tile[0][1],
+                          x1 + (x2 - x1) * (tile[0][0] + tile[1][0]),
+                          y1 + (y2 - y1) * (tile[0][1] + tile[1][1]))
 
-        rpr_image.set_name(image_key)
+        pixels = self.pixels[int(y1): int(y2), int(x1):int(x2), :]
```

Comment:
If image or tile size is small enough the region coordinates would end with 1-pixel width or height, resulting in empty pixels. It will cause the CoreError in the next line.
The [int(y1):int(y2)+1, int(x1):int(x2)+1, :] would fix that.

Reply:
test case: 3*3 image, 800*900 render size, tiles

Reply:
I think adding +1 isn't so correct, that's why added check if y1=y2 or x1=x2 and return None in this case.